Sources/XMLCoder/Auxiliaries/XMLElement.swift

```diff
@@ -0,0 +1,280 @@
+//
+//  XMLElement.swift
+//  XMLCoder
+//
+//  Created by Vincent Esche on 12/18/18.
+//
+
+import Foundation
+
+internal class _XMLElement {
+    static let attributesKey = "___ATTRIBUTES"
+    static let escapedCharacterSet = [("&", "&amp"), ("<", "&lt;"), (">", "&gt;"), ("'", "&apos;"), ("\"", "&quot;")]
+    
+    var key: String
+    var value: String?
+    var attributes: [String: String] = [:]
+    var children: [String: [_XMLElement]] = [:]
+    
+    internal init(key: String, value: String? = nil, attributes: [String: String] = [:], children: [String: [_XMLElement]] = [:]) {
+        self.key = key
+        self.value = value
+        self.attributes = attributes
+        self.children = children
+    }
+    
+    static func createRootElement(rootKey: String, object: ArrayBox) -> _XMLElement? {
```

Comment:
These static functions return the same type they're declared on. This is a good chance to stick to the convention and rewrite this as an optional initializer. This is applicable to all other `createRootElement` functions, which I see delegate to `modifyElement` though. Maybe change to initializers should happen in a separate PR when `_XMLElement` becomes a `struct`?

Reply:
The whole current `createRootElement`/`modifyElement` shenanigans is rather cryptic, imho.

As such I'm all for cleaning up `_XMLElement`.

Reply:
cool, I think `_XMLElement` with these static functions could stay as they are within this PR and change later in a separate one to make the review easier